# HG changeset patch
# User Dan Villiom Podlaski Christiansen <danchr@gmail.com>
# Date 1682100621 -7200
#      Fri Apr 21 20:10:21 2023 +0200
# Branch 1.0.x
# Node ID 9a52223a95e9821b2f2b544ab5a35e06963da3f1
# Parent  2528cff319744f44dc40003cd110700ce4204a94
compat: fix tests with Mercurial default

diff --git a/NEWS.rst b/NEWS.rst
--- a/NEWS.rst
+++ b/NEWS.rst
@@ -1,3 +1,10 @@
+hg-git 1.0.3 (unreleased)
+=========================
+
+This is a minor release, focusing on bugs and compatibility.
+
+* Fix tests with Mercurial 6.5
+
 hg-git 1.0.2 (2023-03-03)
 =========================
 
diff --git a/hggit/gitrepo.py b/hggit/gitrepo.py
--- a/hggit/gitrepo.py
+++ b/hggit/gitrepo.py
@@ -103,7 +103,7 @@
     def known(self):
         raise NotImplementedError
 
-    def peer(self, path=None):
+    def peer(self, path=None, remotehidden=False):
         return self
 
     def stream_out(self):
@@ -136,8 +136,8 @@
 
 # defend against tracebacks if we specify -r in 'hg pull'
 @eh.wrapfunction(hg, b'addbranchrevs')
-def safebranchrevs(orig, lrepo, otherrepo, branches, revs):
-    revs, co = orig(lrepo, otherrepo, branches, revs)
+def safebranchrevs(orig, lrepo, otherrepo, branches, revs, **kwargs):
+    revs, co = orig(lrepo, otherrepo, branches, revs, **kwargs)
     if isinstance(otherrepo, gitrepo):
         # FIXME: Unless it's None, the 'co' result is passed to the lookup()
         # remote command. Since our implementation of the lookup() remote
@@ -245,5 +245,7 @@
         )
 
 
-def make_peer(ui, path, create, intents=None, createopts=None):
+def make_peer(
+    ui, path, create, intents=None, createopts=None, remotehidden=False
+):
     return gitrepo(ui, path, create)
